src/libpriv/rpmostree-postprocess.c

```diff
@@ -220,14 +254,38 @@ do_kernel_prep (GFile         *yumroot,
   gs_unref_object GFile *initramfs_path = NULL;
   const char *boot_checksum_str = NULL;
   GChecksum *boot_checksum = NULL;
-  const char *kname;
-  const char *kver;
+  g_autofree char *kver = NULL;
 
   if (!find_kernel_and_initramfs_in_bootdir (bootdir, &kernel_path,
                                              &initramfs_path,
                                              cancellable, error))
     goto out;
 
+  if (kernel_path == NULL)
+    {
+      gs_unref_object GFile *mod_dir = g_file_resolve_relative_path (yumroot, "usr/lib/modules");
+      gs_unref_object GFile *modversion_dir = NULL;
+
+      if (!find_first_subdirectory (mod_dir, &modversion_dir, cancellable, error))
```

Comment:
I think this could use a comment as to why we're after the 1st subdirectory.  Does 1st imply most recently modified?

Also, is `g_file_enumerate_children()` guaranteed to produce the order you're expecting?